idf_build_apps/app.py

```diff
@@ -804,6 +822,41 @@ def check_should_build(
         self.build_comment = 'current build does not modify any components or files required by this app'
         self._checked_should_build = True
 
+    def check_should_test(self) -> None:
+        """Check if testing is disabled for this app and set test_disable_reason."""
+        if not self.MANIFEST:
+            return
+
+        rule = self.MANIFEST.most_suitable_rule(self.app_dir)
+
+        # Check if testing is enabled for this target
+        enabled_test_targets = rule.enable_test_targets(self.sdkconfig_files_defined_idf_target, self.config_name)
+
+        if self.target not in enabled_test_targets:
+            # Determine the specific reason for test disabling
+            for clause in rule.disable_test:
+                if clause.get_value(self.target, self.config_name or ''):
+                    self.test_comment = f'Disabled by manifest test rule: {clause}'
+                    return
+
+            # Check if disabled by general disable rules
+            for clause in rule.disable:
+                if clause.get_value(self.target, self.config_name or ''):
+                    self.test_comment = f'Disabled by manifest rule: {clause}'
+                    return
```

Comment:
The logic here seems weird. If the logic here is a summary of everything, then build rules should have higher priority than the test rules.

If it's only for the test rules, then the build rules should not be here (caller should merge them together then)